### PR TITLE
[transaction builder generator] add generation of Java transaction builders

### PR DESCRIPTION
**Pull request by matbd**
## Motivation

* Support Java clients
* Add a code example in all supported languages
* Test that the code examples output the expected serialization of a demo transfer script in Libra.
* Update [README](https://github.com/libra/libra/tree/master/language/transaction-builder-generator) with quick start instructions

## Test Plan

```
cargo x test -p transaction-builder-generator
# We don't have Java in Libra's CI.
cargo x test -p transaction-builder-generator -- --ignored
```

## Related PRs

https://github.com/facebookincubator/serde-reflection/pull/27

### CODE REVIEW

File: language/transaction-builder-generator/src/java.rs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+fn output_preamble(out: &mut dyn Write, package: Option<&str>) -> Result<()> {
+    if let Some(name) = package {
+        writeln!(out, "package {};\n", name)?;
+    }
+    writeln!(
+        out,
+        r#"
+import java.math.BigInteger;
+import java.util.Arrays;
+import libra.types.AccountAddress;
```
Comment (thefallentree) [inline language/transaction-builder-generator/src/java.rs:38]: I suggest we use package org.libra.something  here

Comment (matbd) [reply language/transaction-builder-generator/src/java.rs:38]: `org.libra.types` and `org.libra.lcs` make sense but then it gets tricky with `bincode` and `serde`. I need to think about it.

Comment (matbd) [reply language/transaction-builder-generator/src/java.rs:38]: Yes, I think I can work with `org.libra` and `com.facebook`.

Comment (matbd) [reply language/transaction-builder-generator/src/java.rs]: @thefallentree addressed

**Pull request closed without merging**